webanno-api-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/annotation/model/AnnotatorStateImpl.java

```diff
@@ -350,7 +351,17 @@ public void setAnnotationLayers(List<AnnotationLayer> aAnnotationLayers)
         
         // Make sure the currently selected layer is actually visible/exists
         if (!annotationLayers.contains(selectedAnnotationLayer)) {
-            selectedAnnotationLayer = !annotationLayers.isEmpty() ? annotationLayers.get(0) : null;
+            if (!annotationLayers.isEmpty()) {
+                // Make sure that the selected layer is a span layer
+                for (AnnotationLayer layer : annotationLayers) {
```

Comment:
Why not use something like `annotationLayers.stream().filter(is a span type).findFirst().orElse(null)?

Reply:
Because I sometimes forget about the stream functionality in java :(